### ISSUE TITLE
Shared modules across bars

### ISSUE THREAD

**Issue by Alexays** [Alexays/Waybar#368]
Some modules doesn't need to be duplicate across bars.
e.g. `cpu`, `memory`, `idle_inhibitor`, `brightness`, `clock`, `...`
Pb: module config can be different across bars.

Will fix #296

**Comment by b10n** [comment]
Custom modules do not (always) need to be duplicated across bars. I noticed that Waybar 0.9.9 creates a subprocess for each bar. Could this be made configurable?
Perhaps a separate issue is that a subprocess becomes a zombie when a bar is no longer shown, for instance when the output configuration changes.

**Comment by pocek** [comment]
#1276 addresses custom modules.

**Comment by lilydjwg** [comment]
I don't want share clock widgets, because I want it to show seconds on ordinary displays but not on e-ink ones (Constantly changing content is annoying there).

There may also be some other cases sharing is not desired, e.g. different sizes (big monitors vs small ones) or colors (dark theme is terrible for e-ink displays). So sharing (and with which) should be configurable, e.g. define widgets without referring to an output, and then put their ids into each output's configuration.

**Comment by Alexays** [comment]
This is now possible on master via the `include` directive (commit 14f626d4 and follow-ups): factor shared module definitions into a separate file and `"include"` it from each bar, while still overriding per-bar where needed (first-defined wins). Nested includes, wildcards and relative paths are supported. Closing as resolved — please reopen if your sharing use case is not covered.

**Comment by lilydjwg** [comment]
> factor shared module definitions into a separate file and `"include"` it from each bar

I haven't tried yet, but does this handle definition only? So e.g. my custom widget still needs one process per monitor if I want it to appear on every mointor's bars?

**Comment by dakush** [comment]
I think that scripts are still executed multiple times with `include`...

Would it be possible to have a flag in custom modules, so that when it's `true` the script runs only once and the output is sent to all the monitors?

Something like: `"single-instance": <boolean> # (default: false, to keep current behaviour as default)` 

Just a suggestion (I don't know if it's feasible or maybe there's another simpler way), but I think that it would be more efficient (especially for "heavy" scripts") to have the option to run some scripts just once.

Thanks for waybar!
